src/classes/STG_PanelCreatePayments_CTRL.cls

```diff
@@ -62,4 +71,18 @@ public with sharing class STG_PanelCreatePayments_CTRL extends STG_Panel {
         id batchProcessId = database.executeBatch(new PMT_PaymentCreator_BATCH(), 200);
         return null;
     }
+
+    /**
+    * @description Check if the running user has update access to the Opportunity sObject and Read access to the
+    * Payment object
+    * @return Boolean
+    */
+    public Boolean getHasAccess() {
+        String oppObject = Schema.SObjectType.Opportunity.getName();
```

Comment:
It might also be useful to move all of the object declarations like these - for all of the classes - into the `STG_Panel` class as well - as a static var. This way, the call to `Schema.SObjectType.Opportunity.getName()` (and so on) is only made one time for all fo the controllers. That may aid in performance because NPSP Settings loads up every single page at the beginning. As a result, it's effectively making these calls multiple times for every page as it loads its controller.

Reply:
I was reading the dev doc and it seem like token describe is lightweight and execute on compile time. Therefore, I do not think there will be any performance gain to create the static variables in the parent class. [documentation](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_dynamic_describe_objects_understanding.htm#:~:text=Apex%20provides%20two%20data%20structures,is%20used%20for%20token%20describes.)